src/ray/raylet/agent_manager.cc

```diff
@@ -57,7 +57,10 @@ void AgentManager::StartAgent() {
     argv.push_back(arg.c_str());
   }
   argv.push_back(NULL);
-  Process child(argv.data(), nullptr, ec);
+  // Set node id to agent.
+  ProcessEnvironment env;
+  env.insert({"RAY_NODE_ID", options_.node_id.Hex()});
```

Comment:
Just would like to get confirmation this code will be compatible to Windows?

Reply:
I checked that there is a win32 implementation of subprocess env. And raylet use the same way to pass job config to worker, too (please refer to: https://github.com/ray-project/ray/blob/master/src/ray/raylet/worker_pool.cc#L325). I think it will be OK in Windows.